sdk/keyvault/azure_security_keyvault_keys/benches/benchmarks.rs

```diff
@@ -0,0 +1,81 @@
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_keys::{
+    models::{CreateKeyParameters, CurveName, Key, KeyType},
+    KeyClient,
+};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+fn key_operations_benchmark(c: &mut Criterion) {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    // Setup the KeyClient and the CreateKeyParameters
+    async fn setup_key_client() -> (KeyClient, CreateKeyParameters) {
+        let keyvault_url: String = std::env::var("AZURE_KEYVAULT_URL")
+            .unwrap_or_else(|e| panic!("AZURE_KEYVAULT_URL not set: {}", e));
+        let credential = DefaultAzureCredential::new().unwrap();
+        let client: KeyClient = KeyClient::new(&keyvault_url, credential.clone(), None).unwrap();
+        let body = CreateKeyParameters {
+            kty: Some(KeyType::EC),
+            curve: Some(CurveName::P256),
+            ..Default::default()
+        };
+        (client, body)
+    }
+
+    let (client, body) = rt.block_on(async { setup_key_client().await });
+
+    const KEY_NAME: &str = "test-key";
+    // prep key in order to run the benchmark in a clean state
+    rt.block_on(async { create_key(KEY_NAME, &client, body.clone()).await })
+        .unwrap();
```

Comment:
Why not do this in an async closure as well? Yield time to the CPU.

Reply:
because i want to block on this to ensure it is done before the tests start.